pilot/pkg/xds/discovery.go

```diff
@@ -211,7 +211,7 @@ func (s *DiscoveryServer) getNonK8sRegistries() []serviceregistry.Instance {
 	}
 
 	for _, registry := range registries {
-		if registry.Provider() != serviceregistry.Kubernetes || registry.Provider() != serviceregistry.External {
+		if registry.Provider() != serviceregistry.Kubernetes {
```

Comment:
I am just reverting the original PR now basically. ServiceEntry still needs to call UpdateServiceShards. We can probably fix it, but for now doing the minimal fix to avoid regression